### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by Lawmsangi**
### In this pull request, me and my partner implemented the following:
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,5 +8,14 @@
...
   end
+
+  namespace 'api' do
+    namespace 'v1' do
+      resources :posts, only: [:index] do
+        resources :comments, only: [:create, :index]
+      end
+    end
+  end
+
```
Comment (danmainah) [inline config/routes.rb]: - [ ] @Lawmsangi and @lilskyex0x Great work adding code to add and display comments and also display posts, however, I found it hard creating a new comment as it requires authentication, which you have not provided a way to add in the API, kindly consider making it possible to add new comment.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Lawmsangi and @lilskyex0x , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Created an API endpoint to list all posts for a user.
✔️  Created an API endpoint to list all comments for a user's post and add new comments.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Lawmsangi** [comment]
I disabled authentication since devise does not support token based authentication from version 4 onwards and the project requirements are no longer possible to achieve ( it needs to be updated) 
For api based apps gems devise-token-auth or devise jwt should be used and not plain devise

**Comment by Lawmsangi** [comment]
I disabled authentication since devise does not support token based authentication from version 4 onwards and the project requirements are no longer possible to achive ( it needs to be updated) 
For api based apps gems devise-token-auth or devise jwt should be used and not plain devise

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    render json: @comments
+  end
+
+  def create
+    user_id = params[:user_id]
+    post_id = params[:post_id]
+    @comment = Comment.new(user_id:, post_id:, text: comment_params[:text])
+
+    if @comment.save
+      render json: 'Comment saved successfully!'
```
Comment (shahadat3669) [inline app/controllers/api/v1/comments_controller.rb]: 👍 Good effort on API functionality! 👍 However, there's a minor adjustment needed to meet project requirements:

- [ ] **Valid JSON Response:** While your API endpoints correctly receive JSON, the response currently is not valid JSON. I recommend updating your code to ensure that the API endpoints respond with valid JSON according to project requirements.
image

By ensuring that the response is valid JSON, you'll align with project specifications for API functionality. Keep up the good work! 🚀📊

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @lilskyex0x, @Lawmsangi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Everyone contributed
✅ Worked on previous reviewer comments.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi @Lawmsangi, @lilskyex0x , 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No linter Errors
✅ work on the previous reviewer comment.
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**